README.md

```diff
@@ -5,154 +5,85 @@ Swiftz
-let user: User? = JSValue.decode(userjs) >>- User.fromJSON
-XCTAssert(user! == User("max", 10, ["hello"], ["one": "1"]))
-```
-
-**Concurrency:**
-
-```swift
-// we can delay computations with futures
-let x: Future<Int> = Future(exec: gcdExecutionContext, {
-	sleep(1)
-	return 4
-})
-x.result() == x.result() // true, returns in 1 second
-
-// Channels
-let chan: Chan<Int> = Chan()
-chan.write(1)
-chan.write(2) // this could happen asynchronously
-let x1 = chan.read()
-let x2 = chan.read()
-println((x1, x2)) // 1, 2
-
-// we can map and flatMap over futures
-x.map({ $0.description }).result() // "4", returns instantly
-x.flatMap({ (x: Int) -> Future<Int> in
-	return Future(exec: gcdExecutionContext, { sleep(1); return x + 1 })
-}).result() // sleeps another second, then returns 5
-```
-
-Swiftz Core
------------
-
-Swiftz Core only contains interfaces to concepts and implementations
-for data abstractions from the standard library.
+It defines functional data structures, functions, idioms, and extensions that augment 
+the Swift standard library.
+
+For a small, simpler way to introduce functional primitives into any codebase,
+see [Swiftx](https://github.com/typelift/Swiftx). 
+
+Setup
+-----
+
+To add Swiftz to your application:
+
+**Using Carthage**
+
+- Add Swiftz to your Cartfile
+- Run `carthage update`
+- Drag the relevant copy of Swiftz into your project.
+- Expand the Link Binary With Libraries phase
+- Click the + and add Swiftz
+- Click the + at the top left corner to add a Copy Files build phase
+- Set the directory to `Frameworks`
+- Click the + and add Swiftz
+
+**Using Git Submodules**
+
+- Clone Swiftz as a submodule into the directory of your choice
+- Run `git submodule init -i --recursive`
+- Drag `Swiftz.xcodeproj` or `Swiftz-iOS.xcodeproj` into your project tree as a subproject
+- Under your project's Build Phases, expand Target Dependencies
+- Click the + and add Swiftz
+- Expand the Link Binary With Libraries phase
+- Click the + and add Swiftz
+- Click the + at the top left corner to add a Copy Files build phase
+- Set the directory to `Frameworks`
+- Click the + and add Swiftz
+
+Introduction
+------------
+
+Swiftz draws inspiration from a number of functional libraries 
+and languages.  Chief among them are [Scalaz](https://github.com/scalaz/scalaz),
+[Prelude/Base](https://hackage.haskell.org/package/base), [SML
+Basis](http://sml-family.org/Basis/), and the [OCaml Standard
+Library](http://caml.inria.fr/pub/docs/manual-ocaml/stdlib.html).  Elements of
+the library rely on their combinatorial semantics to allow declarative ideas to
+be expressed more clearly in Swift.
+
+Swiftz is a proper superset of [Swiftx](https://github.com/typelift/Swiftx) that
+implements higher-level data types like Lenses, Zippers, HLists, and a number of
+typeclasses integral to programming with the maximum amount of support from the
+type system.  A small number of examples are included in the test suite to
```

Comment:
Do you have anymore suggestions for examples?   We don't have anything related to Zippers, Concurrency, the Lens Family (besides a few small methods), and the typeclasses